kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/model/ProxyModelBuilder.java

```diff
@@ -36,6 +36,9 @@ public ProxyModelBuilder(DependencyResolver resolver) {
 
     public ProxyModel build(KafkaProxy primary, Context<KafkaProxy> context) {
         ProxyResolutionResult resolutionResult = resolver.resolveProxyRefs(primary, context);
+        if (!resolutionResult.allClusterDependenciesReconciled()) {
+            throw new IllegalStateException("Dependencies not fully reconciled");
```

Comment:
I would be inclined to use a dedicated exception type for this.

Reply:
Should also extend the message to talk about how it should be a transient state